src/components/Image/index.tsx

```diff
@@ -1,24 +1,22 @@
-import lodashGet from 'lodash/get';
 import React, {useEffect, useMemo} from 'react';
 import {Image as RNImage} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
 import ONYXKEYS from '@src/ONYXKEYS';
-import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
+import {ImageOnyxProps, ImageProps, ImagePropsWithOnyx} from './types';
 
-function Image(props) {
-    const {source: propsSource, isAuthTokenRequired, onLoad, session} = props;
+function Image({source: propsSource, isAuthTokenRequired, onLoad, session, ...forwardedProps}: ImagePropsWithOnyx) {
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      */
     const source = useMemo(() => {
-        if (isAuthTokenRequired) {
-            // There is currently a `react-native-web` bug preventing the authToken being passed
-            // in the headers of the image request so the authToken is added as a query param.
-            // On native the authToken IS passed in the image request headers
-            const authToken = lodashGet(session, 'encryptedAuthToken', null);
+        // There is currently a `react-native-web` bug preventing the authToken being passed
+        // in the headers of the image request so the authToken is added as a query param.
+        // On native the authToken IS passed in the image request headers
+        const authToken = session?.encryptedAuthToken ?? null;
+
+        if (isAuthTokenRequired && authToken && typeof propsSource !== 'number' && propsSource.uri) {
```

Comment:
I'm noticing a lot of subtle changes on the logic of this component in this PR, let's make sure it's very well tested and avoid risk of regressions.

Reply:
Sure thing, I will do a lot of testing once we manage the syntax stuff 😄